js/ui/appDisplay.js

```diff
@@ -493,18 +497,21 @@ class AppDisplay extends BaseAppView {
             if (!this._currentDialog.contains(actor))
                 this._currentDialog.popdown();
         });
-        Main.overview.addAction(this._clickAction, false);
-        this._eventBlocker.bind_property('reactive', this._clickAction,
-            'enabled', GObject.BindingFlags.SYNC_CREATE);
-
-        this._bgAction = new Clutter.ClickAction();
-        Main.overview.addAction(this._bgAction);
-        BackgroundMenu.addBackgroundMenuForAction(this._bgAction, Main.layoutManager);
-        this._clickAction.bind_property('enabled', this._bgAction,
-            'enabled',
-            GObject.BindingFlags.SYNC_CREATE | GObject.BindingFlags.INVERT_BOOLEAN);
-        this.bind_property('mapped', this._bgAction,
-            'enabled', GObject.BindingFlags.SYNC_CREATE);
+        if (params.addBackgroundAction) {
+            Main.overview.addAction(this._clickAction, false);
+            this._eventBlocker.bind_property('reactive', this._clickAction,
+                'enabled', GObject.BindingFlags.SYNC_CREATE);
+        }
+
+        this._bgAction = new Clutter.ClickAction({
+            long_press_duration: MENU_POPUP_TIMEOUT * 2,
```

Comment:
Wouldn't this be too long? 1.2s to click and hold to show background menu?

Reply:
Maybe 1.5x instead of 2x or lower the menu popup timeout? Fwiw, the default long-press timeout on clutter is 500.